### PR TITLE
Add test to ensure that org admins are never team members

### PR DESCRIPTION
**Pull request by cblecker**
Org admins are always listed as team maintainers in the API. This ensures that an org admin is never listed as just a member.. they can either be a maintainer, or not on the team at all.

### CODE REVIEW

**Comment by cblecker** [comment]
working as intended:
```
==================== Test output for //config:go_default_test:
--- FAIL: TestAllOrgs (0.05s)
    config_test.go:242: The team kubernetes-milestone-maintainers has org admins listed as members; these users should be in the maintainers list instead: cblecker, fejta, idvoretskyi
FAIL
================================================================================
```

**Comment by cblecker** [comment]
/assign @fejta 
/cc @justaugustus 
/hold
Holding for @fejta's review of the tests 🤖

File: config/config_test.go [hunk 1]
```diff
@@ -131,8 +131,13 @@ func testTeamMembers(teams map[string]org.Team, orgMembers sets.String) []error
 			errs = append(errs, fmt.Errorf("The following members of team %s are not org members: %s", teamName, strings.Join(missing.List(), ", ")))
 		}
 
+		// check if admins are a regular member of team
+		if adminTeamMembers := teamMembers.Intersection(admins); len(adminTeamMembers) > 0 {
+			errs = append(errs, fmt.Errorf("The team %s has org admins listed as members; these users should be in the maintainers list instead: %s", teamName, strings.Join(adminTeamMembers.List(), ", ")))
```
Comment (fejta) [inline config/config_test.go]: Or removed?

**Comment by justaugustus** [comment]
Awesome!

(for the SIG Release stuff)
/lgtm
/approve

**Pull request merged**